src/wikia-sharing.plugin.js

```diff
@@ -0,0 +1,317 @@
+		if (!this.wikiaSharingElement.style.display) {
+			this.open();
+		} else {
+			this.close();
+		}
+	}.bind(this), this.buttonID, 'wikia-jw-sharing-button');
+};
+
+wikiaJWPlayerSharingPlugin.prototype.removeButton = function () {
+	this.player.removeButton(this.buttonID);
+};
+
+/**
+ * closes sharing menu
+ */
+wikiaJWPlayerSharingPlugin.prototype.close = function () {
+	this.container.style.display = null;
+	this.player.getContainer().classList.remove('wikia-jw-sharing-open');
+};
+
+/**
+ * opens sharing menu
+ */
+wikiaJWPlayerSharingPlugin.prototype.open = function () {
+	this.wikiaSharingElement.innerHTML = '';
+	this.wikiaSharingElement.appendChild(this.getContent());
+	showElement(this.container);
+	this.player.trigger('wikiaShareMenuExpanded');
+	this.player.getContainer().classList.add('wikia-jw-sharing-open');
+};
+
+/**
+ * hides the entire plugin (button and sharing menu)
+ */
+wikiaJWPlayerSharingPlugin.prototype.hide = function () {
+	this.close();
+	this.removeButton();
+};
+
+/**
+ * shows back the entire plugin (adds button back)
+ */
+wikiaJWPlayerSharingPlugin.prototype.show = function () {
+	if (!this.getSharingButtonElement()) {
+		this.addButton();
+	}
+};
+
+wikiaJWPlayerSharingPlugin.prototype.getContent = function () {
+	var fragment = document.createDocumentFragment(),
+		header = document.createElement('h3');
+
+	header.innerText = this.config.i18n.sharing;
+	fragment.appendChild(header);
+	fragment.appendChild(this.createSharingListElement());
+
+	return fragment;
+};
+
+wikiaJWPlayerSharingPlugin.prototype.addSharingContent = function (div) {
+	div.classList.add('wikia-jw-sharing');
+	div.classList.remove('jw-reset');
+	div.classList.remove('jw-plugin');
+
+	this.show();
+};
+
+wikiaJWPlayerSharingPlugin.prototype.createSharingListElement = function () {
+	var sharingList = document.createElement('ul');
+
+	sharingList.className = 'wikia-jw-sharing__list wds-list';
+
+	var userLang = this.getUserLang(),
+		socialNetworks = this.socialNetworks[userLang];
+
+	if (socialNetworks) {
+		socialNetworks.forEach(function (socialNetwork) {
+			sharingList.appendChild(this.getSocialNetworkButton(socialNetwork));
+		}.bind(this));
+	}
+
+	return sharingList;
+};
+
+wikiaJWPlayerSharingPlugin.prototype.getSocialNetworkButton = function (socialNetwork) {
+	var button = document.createElement('button');
+
+	button.className = 'wds-is-square wds-is-' + socialNetwork + '-color wds-button';
+	button.appendChild(createSVG(wikiaJWPlayerIcons[socialNetwork]));
+	button.addEventListener('click', function () {
+		this.player.trigger('socialNetworkClicked', {socialNetwork: socialNetwork});
+		window.open(this[socialNetwork]());
+	}.bind(this));
+
+	return button;
+};
+
+wikiaJWPlayerSharingPlugin.prototype.getUserLang = function() {
+	return (window.navigator.userLanguage || window.navigator.language).slice(0, 2);
+};
+
+wikiaJWPlayerSharingPlugin.prototype.socialNetworks = {
```

Comment:
do we have a ticket to extract this array to design system or any other shared place? and to use this shared place in all clients?

Reply:
Yea, I was planning to create one for x-force. Will do.